src/Actions/Product/CreateProduct.php

```diff
@@ -54,6 +56,13 @@ public function performAction(): Product
             $product->prices()->createMany($prices);
         }
 
+        if ($productCrossSellings) {
+            foreach ($productCrossSellings as $productCrossSelling) {
+                $productCrossSelling['product_id'] = $product->id;
+                CreateProductCrossSelling::make($productCrossSelling)->checkPermission()->validate()->execute()->id;
```

Comment:
You know that this will eventually throw an exception. This may cause some confusion, because the product is created, but an exception is thrown. Rethink if you need to validate. Also the permission should be checked before anything happens with the product to avoid confusion.